test/conformance/context/urContextGetInfo.cpp

```diff
@@ -56,6 +57,34 @@ TEST_P(urContextGetInfoTestWithInfoParam, Success) {
     std::vector<uint8_t> info_data(info_size);
     ASSERT_SUCCESS(
         urContextGetInfo(context, info, info_size, info_data.data(), nullptr));
+
+    switch (info) {
+    case UR_CONTEXT_INFO_NUM_DEVICES: {
+        auto returned_num_of_devices =
+            reinterpret_cast<uint32_t *>(info_data.data());
+        ASSERT_EQ(*returned_num_of_devices,
+                  uur::DevicesEnvironment::instance->devices.size());
+        break;
+    }
+    case UR_CONTEXT_INFO_DEVICES: {
+        auto returned_devices =
+            reinterpret_cast<ur_device_handle_t *>(info_data.data());
+        for (uint32_t i = 0;
+             i < uur::DevicesEnvironment::instance->devices.size(); i++) {
```

Comment:
I'd be surprised if this broke any time soon but I don't think we make any guarantees about the order of devices returned for this query, and this test is now enforcing that. We already verify that the number of devices is what we expect in another test, so I think it would be fine to just verify that we can find each device from `uur::DevicesEnvironment::instance->devices` in the output.

Reply:
Thanks for raising this point, you are right. I also changed that to address the fact that we could create a context with not all the available devices.